File: sdk/go/pkg/replay/runner.go
package replay

import (
	"context"
	"fmt"

	"github.com/evoreplay/go-sdk/pkg/adapter"
)

// Run executes the given entrypoint in Replay Mode
func Run(ctx context.Context, artifactData []byte, entryPoint func(context.Context)) error {
	rec := adapter.Get()
	
	fmt.Println("[EvoReplay] Loading Artifact...")
	if err := rec.LoadArtifact(artifactData); err != nil {
		return err
	}
	
	fmt.Println("[EvoReplay] Replaying...")
	entryPoint(ctx)
	
	fmt.Println("[EvoReplay] Done.")
	return nil
}

File: sdk/node/src/replay/runner.js
/**
 * Replay Runner
 * Orchestrates the replay of an artifact.
 */
const { recorder } = require('../adapter/recorder');
const { init } = require('../index');

class ReplayRunner {
    constructor(artifact) {
        this.artifact = typeof artifact === 'string' ? JSON.parse(artifact) : artifact;
    }

    /**
     * Run the replay
     * @param {Function} entryPoint - The application function to re-execute
     */
    async run(entryPoint) {
        console.log('[EvoReplay] Starting Replay Session...');

        // 1. Initialize SDK (Hooks)
        init();

        // 2. Load Artifact
        recorder.loadArtifact(this.artifact);

        // 3. Execute
        try {
            console.log('[EvoReplay] Executing application logic in REPLAY mode.');
            await entryPoint();
        } catch (e) {
            console.error('[EvoReplay] Replay Execution Error:', e);
        }

        console.log('[EvoReplay] Replay Finished.');
    }
}

module.exports = { ReplayRunner };
